src/assignment/modals/AddCollectionModal.tsx

```diff
@@ -0,0 +1,368 @@
+import {
+	Button,
+	ButtonGroup,
+	ButtonToolbar,
+	Flex,
+	FlexItem,
+	Form,
+	FormGroup,
+	Icon,
+	Modal,
+	ModalBody,
+	Spacer,
+	Table,
+	TableColumn,
+	TextInput,
+	Toggle,
+	Toolbar,
+	ToolbarItem,
+	ToolbarLeft,
+	ToolbarRight,
+} from '@viaa/avo2-components';
+import { Avo } from '@viaa/avo2-types';
+import { noop } from 'lodash-es';
+import React, { FunctionComponent, useCallback, useEffect, useMemo, useState } from 'react';
+import { useTranslation } from 'react-i18next';
+
+import { CollectionService } from '../../collection/collection.service';
+import { ContentTypeNumber } from '../../collection/collection.types';
+import { LoadingErrorLoadedComponent, LoadingInfo } from '../../shared/components';
+import { CustomError, formatDate, formatTimestamp, isMobileWidth } from '../../shared/helpers';
+import { getOrderObject } from '../../shared/helpers/generate-order-gql-query';
+import { truncateTableValue } from '../../shared/helpers/truncate';
+import { useTableSort } from '../../shared/hooks';
+import { ToastService } from '../../shared/services';
+import i18n from '../../shared/translations/i18n';
+import { TableColumnDataType } from '../../shared/types/table-column-data-type';
+import { AssignmentOverviewTableColumns } from '../assignment.types';
+
+// Column definitions
+const GET_ADD_COLLECTION_COLUMNS = (): TableColumn[] => [
+	{
+		id: 'title',
+		label: i18n.t('assignment/modals/add-collection-modal___titel'),
+		sortable: true,
+		dataType: TableColumnDataType.string,
+	},
+	{
+		id: 'updated_at',
+		label: i18n.t('assignment/modals/add-collection-modal___laatst-bewerkt'),
+		sortable: true,
+		dataType: TableColumnDataType.dateTime,
+	},
+	{
+		id: 'is_public',
+		label: i18n.t('assignment/modals/add-collection-modal___is-publiek'),
+		sortable: true,
+		dataType: TableColumnDataType.boolean,
+	},
+];
+
+const TABLE_COLUMN_TO_DATABASE_ORDER_OBJECT: Partial<{
+	[columnId in keyof Avo.Collection.Collection]: (order: Avo.Search.OrderDirection) => any;
+}> = {
+	title: (order: Avo.Search.OrderDirection) => ({
+		title: order,
+	}),
+	updated_at: (order: Avo.Search.OrderDirection) => ({
+		updated_at: order,
+	}),
+	is_public: (order: Avo.Search.OrderDirection) => ({
+		is_public: order,
+	}),
+};
+
+export interface AddCollectionModalProps {
+	user?: Avo.User.User;
```

Comment:
you can avoid having to pass user as an object by using the withUser higher order component. For one property this is optional, but for multiple eg: (history, location, match, user) it is easier to use the higher order components.

example:
```typescript
interface HomeProps {
   title: string;
}

const Home: FunctionComponent<HomeProps & UserProps & RouteComponentProps> = ({ history, user }) => {
	...
};

export default compose(withRouter, withUser)(Home) as FunctionComponent<HomeProps>;
```

if you only need user and not router related props:

```typescript
interface HomeProps {
   title: string;
}

const Home: FunctionComponent<HomeProps & UserProps> = ({ history, user }) => {
	...
};

export default withUser(Home) as FunctionComponent<HomeProps>;
```

https://github.com/viaacode/avo2-client/blob/b2658fe3d58e65304dc87567ce9f417484e9d2d7/src/home/views/Home.tsx#L15


https://github.com/viaacode/avo2-client/blob/b2658fe3d58e65304dc87567ce9f417484e9d2d7/src/home/views/Home.tsx#L39

The advantage here is that devs that use your component don't need to pass user:
```typescript
<Home title="test" />
```